### PR TITLE
[CHE-138] Clean Up Dependencies

### PR DESCRIPTION
**Pull request by seantokuzo**
### Description

- Moves dependencies to relevant package.json files (client vs. server)
- Moves .babelrc to client folder
- Updates dependencies to latest versions (updated package-lock.json)
- Updated syntax for setting proxy in webpack devServer (bc of updated webpack package)
- Integrates postcss-loader for use with Tailwind CSS
- Creates postcss.config.js to configure postcss for Tailwind CSS
- Updated css import statement in index.tsx -> output.css no longer needed
- Updated deprecated mongoose syntax in userController.ts and useController.tests.ts (findOneAndRemove -> findOneAndDelete)
- Updated mockExit type in db.test.ts that was causing test to fail

### Jira Task

[CHE-138](https://codehammers.atlassian.net/browse/CHE-138?atlOrigin=eyJpIjoiZjFmMzZkYjkwYmFkNDMzMDlmYTYwYThlMzJkOGJkMmEiLCJwIjoiaiJ9)

### Testing Instructions

- Run `npm run test:all` and make sure tests are passing
- Run application with `npm run dev-ts` and make sure app spins up and functions properly
- Rebuild docker image and run `npm run docker-dev` to make sure app functions properly in dockerized environment

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [x] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recommended style guide.

### Additional Notes, Images, etc.

This PR makes major changes to dependencies - will require a rebuild of docker images

Please double check that tests are passing and application functions as expected (don't let my first PR blow everything up 🙏)

[CHE-138]: https://codehammers.atlassian.net/browse/CHE-138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/src/index.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import React from "react";
 import ReactDOM from "react-dom/client";
 import { Provider } from "react-redux";
 import { store } from "./app/store";
-import "../dist/output.css";
+// import "../dist/output.css";
```
Comment (JimmyTran48) [inline client/src/index.tsx:5]: Do we need?

File: __tests__/db.test.ts [hunk 2]
```diff
@@ -10,14 +10,14 @@ jest.mock("mongoose", () => ({
 }));
 
 describe("connectDB", () => {
-  let mockExit: jest.SpyInstance<void, [code?: number | undefined]>;
+  let mockExit: jest.SpyInstance<void, [code?: string | number | undefined | null]>;
```
Comment (JimmyTran48) [inline __tests__/db.test.ts:13]: There a way to consolidate the type for `code`

File: client/src/index.tsx (hunk 1 above)
Comment (seantokuzo) [reply client/src/index.tsx:5]: Nostalgic value?

**Review by JimmyTran48 — CHANGES_REQUESTED** [review]
Left some in line comments

**Comment by seantokuzo** [comment]
Gotta look into client node_modules not being populated into the test container

**Review by JimmyTran48 — APPROVED** [review]

File: __tests__/db.test.ts (hunk 2 above)
Comment (seantokuzo) [reply __tests__/db.test.ts]: I think this was resolved? Removing all the typing allowed it to infer the proper type for "code"

**Review by brok3turtl3 — APPROVED** [review]
Great work @seantokuzo!

**Review by brok3turtl3 — CHANGES_REQUESTED** [review]
Hold that thought - when I spin things up in a test dev container with a new image based of this branch I get
Screenshot 2024-05-13 at 9 19 50 AM
Screenshot 2024-05-13 at 9 19 36 AM

**Review by brok3turtl3 — APPROVED** [review]
Docker flow fixed - Everything looks good.

**Pull request merged**